packages/@aws-cdk/aws-lambda-destinations/lib/lambda.ts

```diff
@@ -1,22 +1,65 @@
+import * as events from '@aws-cdk/aws-events';
+import * as targets from '@aws-cdk/aws-events-targets';
 import * as lambda from '@aws-cdk/aws-lambda';
 import { Construct } from '@aws-cdk/core';
+import { EventBridgeDestination } from './event-bridge';
+
+/**
+ * Options for a Lambda destination
+ */
+export interface LambdaDestinationOptions {
+  /**
+   * Whether the destination function receives only the `responsePayload` of
+   * the origin function.
+   *
+   * @default false The destination function receives the full invocation record.
+   */
+  readonly responseOnly?: boolean;
+}
 
 /**
  * Use a Lambda function as a Lambda destination
  */
 export class LambdaDestination implements lambda.IDestination {
-  constructor(private readonly fn: lambda.IFunction) {
+  constructor(private readonly fn: lambda.IFunction, private readonly options: LambdaDestinationOptions = {}) {
   }
 
   /**
    * Returns a destination configuration
    */
-  public bind(_scope: Construct, fn: lambda.IFunction): lambda.DestinationConfig {
-    // deduplicated automatically
-    this.fn.grantInvoke(fn);
+  public bind(scope: Construct, fn: lambda.IFunction, options?: lambda.DestinationOptions): lambda.DestinationConfig {
+    // Normal Lambda destination (full invocation record)
+    if (!this.options.responseOnly) {
+      // deduplicated automatically
+      this.fn.grantInvoke(fn);
+
+      return {
+        destination: this.fn.functionArn
+      };
+    }
+
+    // Otherwise add rule to extract the response payload and use EventBridge
+    // as destination
+    if (!options) { // `options` added to bind() as optionnal to avoid breaking change
+      throw new Error('Options must be defined when using `responseOnly`.');
+    }
+
+    // Match invocation result of the function to which this destination is added and
+    // use it to trigger destination function
+    new events.Rule(scope, options.type, {
+      eventPattern: {
+        detailType: [`Lambda Function Invocation Result - ${options.type}`],
+        resources: [`${fn.functionArn}:$LATEST`],
+        source: ['lambda'],
+      },
+      targets: [
+        new targets.LambdaFunction(this.fn, {
+          event: events.RuleTargetInput.fromEventPath('$.detail.responsePayload')
```

Comment:
What is the value of `$.detail.responsePayload` in the case of failure? 

Going by https://docs.aws.amazon.com/lambda/latest/dg/invocation-async.html#invocation-async-destinations, it seems like the responsePayload contains only the error message and that too the first 1KB of it, yes? 

Does it still make sense to unwrap here - in this case, it probably makes more sense to look at responseContext and responsePayload?

Reply:
> What is the value of `$.detail.responsePayload` in the case of failure?

The error object returned by the Lambda function.

For me it still makes sense to use this as the event. Or maybe ``$.detail.responsePayload.errorMessage`? `responseContext` doesn't give much information (`functionError` will be either `Handled` or `Unhandled`).

I seems that the 1KB limit applies only for dead letter queues, not for destinations.

Reply:
Ok, let's lean towards consistency. Leave it as is. But also, let's make it clear in the README and in this option's documentation that this only unwraps the `detail` structure both in the case of success and failure.

It would be clearer to put one example of each in the tsdoc for this option.